Answers.md

```diff
@@ -0,0 +1,9 @@
+Explain the differences between client-side routing and server-side routing.  :: Server-side gets updated on a server and persists on that server until the server is killed - therefore client-side routing will persist even after refreshes to the page being changed/updated, utilizing memory.  Client-side exists locally, and will reset every time the page is refreshed.
```

Comment:
**server-side routing:** User clicks on a link. The browser makes HTTP GET request to the URL in the `href`. The server processes the request, sends a new HTML document as response. The browser discards previous web page and displays the new one:

**client-side routing:** User clicks on a link. Client-side code (routing library) catches the event, detects the URL, and prevents browser from default behavior (making HTTP GET request). The routing library then manually changes the URL displayed in the browser (using HTML5 history API). Routing library then changes the state of the client app (rendering specific React component).ng library then changes the state of the client app (rendering specific React component).